### PR TITLE
Keymgr dpe dv xfer pr

### PR DESCRIPTION
**Pull request by bobby-rivos**
This PR contains commits that fix the scoreboard checkers and base seq checkers for:

Key slot modeling
key_policy modeling
error checking
sw/hw generation
The PR also creates the smoke test for keymgr_dpe_smoke_vseq.sv.

writes to all key-slots
test allow_child
tests retain_parent and non retain_parent
test boot_stage
test max_key_version
With this PR keymgr_dpe will have basic checking enabled and a smoke test with functioning checks.

### CODE REVIEW

**Review by ballifatih — APPROVED** [review]
I have reviewed this PR on the other repo https://github.com/lowRISC/opentitan-integrated/pull/686, and I dropped few comments. Thanks @bobby-rivos.

**Comment by bobby-rivos** [comment]
@ballifatih just noticed these. I'll go ahead and address some of those changes as a lot of them are relatively small tweaks or clarifications. Thanks for the effort in your review!

**Comment by bobby-rivos** [comment]
I addressed the comments on https://github.com/lowRISC/opentitan-integrated/pull/686 can I get a re-approval on this PR? Thanks!

**Review by ballifatih — APPROVED** [review]
LGTM, thanks for addressing comments in the other PR. I think there are still few moving pieces in the scoreboard, so those comments on the other PR are not a blocker for this PR, but rather discussion points for the future fixes.

I think you need an approval from a committer too. Thank you @bobby-rivos.

File: hw/ip/keymgr_dpe/dv/env/keymgr_dpe_env.sv
Comment (jdonjdon) [inline hw/ip/keymgr_dpe/dv/env/keymgr_dpe_env.sv:28]: This has been added by cip_base_env. See [here](https://github.com/lowRISC/opentitan/blob/6ed82f97abb2f9977a1e1957f27d274b05354ba4/hw/dv/sv/cip_lib/cip_base_env.sv#L27-L31)
You don't need it.

**Review by jdonjdon — COMMENTED** [review]
Please see the comment

File: hw/ip/keymgr_dpe/dv/env/keymgr_dpe_env.sv
Comment (bobby-rivos) [reply hw/ip/keymgr_dpe/dv/env/keymgr_dpe_env.sv]: Acknowledged and this code was removed.

**Review by jdonjdon — APPROVED** [review]

**Pull request merged**
